### ISSUE TITLE
README.md: move out some content

### ISSUE THREAD

**Issue by AlexGuteniev** [microsoft/STL#2341]
>As an outside (but somewhat related) observation, I wonder if it would be beneficial to move some of the content from the README.md into this CONTRIBUTING.md instead.  The README's length continues to grow and I wonder if we are reaching a point of diminishing returns where the README is reaching a size where it's difficult to parse. 
>
>I propose moving the following sections to CONTRIBUTING.md since they seem to be directly related to proposing/reviewing changes.  
>* How To Build With (The Visual Studio IDE | A Native Tools Command Prompt)
>* How to Consume
>* How To Run The Tests With A Native Tools Command Prompt
>* (potentially) Editing And Testing The Debugger Visualizer
>
>I also recognize that this may be out of the scope of this PR and I am also happy to have a tracking issue filed instead.
>
_Originally posted by @mnatsuhara in https://github.com/microsoft/STL/pull/2212/files#r750639997_

I'm not sure if build instructions should be moved out from `README.md`
As an alternative, I'm proposing move out issue reporting instructions instead.

**Comment by AlexGuteniev** [comment]
I was opposed to this even back then when this suggestion was made. Now I'm even more confident that we shouldn't do this.

README.md has grew since from 509 lines (397 loc) · 28 KB to 592 lines (467 loc) · 32.2 KB. Not very fast. I've contributed to the recent growth by adding "testing with ASan" part. I didn't feel any editing inconveniences caused by too big file. Apparently browsers/scripts capabilities to edit large files grow faster than this file.

If we are still to extract something, we should consider extracting things that are not something every significant use of the repo would need. So I don't think building/consuming/testing should be extracted. I'd consider:
 * Benchmarking -- not needed for most PRs. We could move out to the existing wiki more.
 * Running tests with ASan -- effectively only needed for vectorization PRs. Can go to a separate doc. The doc could possibly be dug as deep as into `/docs/` next to `import_library.md`.

(Editing And Testing The Debugger Visualizer could also have been a candidate for extraction, but it is too short to make its extraction reasonable)

**Comment by StephanTLavavej** [comment]
I agree, I think that splitting up the README would make information harder to find, not easier. There's even a built-in table of contents.
